src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -725,6 +727,19 @@ void IMergeTreeDataPart::loadTTLInfos()
     }
 }
 
+void IMergeTreeDataPart::loadFingerprint()
+{
+    String path = getFullRelativePath() + "fingerprint.txt";
+    if (volume->getDisk()->exists(path))
+    {
+        auto in = openForReading(volume->getDisk(), path);
+
+        readText(fingerprint, *in);
+        if (fingerprint.empty())
+            throw Exception("Found fingerprint.txt but it is empty in part: " + name, ErrorCodes::LOGICAL_ERROR);
```

Comment:
Not a logical error. `DATA_CORRUPTION`?

Reply:
more like that, the PR is based upon whatever is agreed here #14860